Mathlib/LinearAlgebra/PerfectPairing.lean

```diff
@@ -57,7 +59,61 @@ protected def flip : PerfectPairing R N M where
 
 @[simp] lemma flip_flip : p.flip.flip = p := rfl
 
--- TODO `M` and `N` are both reflexive
+/-- The linear equivalence from `M` to `Dual R N` induced by a perfect pairing. -/
+@[simps]
+noncomputable def toDualLeft : M ≃ₗ[R] Dual R N where
+  toFun := p.toLin
+  map_add' := LinearMap.map_add p.toLin
+  map_smul' := LinearMapClass.map_smul p.toLin
+  invFun f := (bijective_iff_has_inverse.mp p.bijectiveLeft).choose f
+  left_inv := (bijective_iff_has_inverse.mp p.bijectiveLeft).choose_spec.1
+  right_inv := (bijective_iff_has_inverse.mp p.bijectiveLeft).choose_spec.2
+
+theorem toDualLeft_invFun (f : Dual R N) (x : N) : p.toLin (p.toDualLeft.invFun f) x = f x := by
```

Comment:
```suggestion
theorem toDualLeft_invFun (f : Dual R N) (x : N) : p ((p.toDualLeft).symm f) x = f x := by
```
We usually don't use `invFun` because it is merely a function and not a linear map / linear equiv. 
Maybe you can call it `apply_toDualLeft_symm_apply` etc.

You might also want to make it a simp lemma.

Reply:
similarly below